samples/tab-meeting-transcript-recording/nodejs/src/components/meetingtranscriptrecording.js

```diff
@@ -79,11 +139,71 @@ const MeetingTranscriptRecording = () => {
                           setIsLoginVisible(false);
                           setLoading(false);
                           setIsCardVisible(true);
+                          setEventUpdated(false);
+                          data.map((event, index) => (
+                            getMeetingTranscriptsIdRecordingId(clientSideToken, event.joinUrl)
+                          ))
                         } else {
                             setLoading(false);
                             reject(response.error);
                             setIsConsentButtonVisible(true);
                             setIsLoginVisible(false);
+                            setEventUpdated(false);
+                        }
+                    })
+            });
+        });
+    }
+
+    // Get server side token and user profile.
+    const getMeetingTranscriptsIdRecordingId = (clientSideToken, joinUrl) => {
+        return new Promise((resolve, reject) => {
+            microsoftTeams.app.getContext().then((context) => {
+                fetch(`/getMeetingTranscriptsIdRecordingId?joinUrl=${joinUrl}&ssoToken=${clientSideToken}`, {
+                    method: 'get',
+                    headers: {
+                        "Content-Type": "application/text",
+                        "Authorization": "Bearer " + clientSideToken
+                    },
+                    cache: 'default'
+                })
+                .then(response => response.json())
+                .then(data => {                    
+                        if(data.error=="consent_required"){
+                            setIsConsentButtonVisible(true);
+                        }
+                        if (data) {
+                            setCardData(data);
+                        } else {
+                            reject(response.error);
+                        }
+                    })
+            });
+        });
+    }
+
+    const createSubscription = (clientSideToken) => {
+        return new Promise((resolve, reject) => {
+            microsoftTeams.app.getContext().then((context) => {
+                fetch('/createsubscription?ssoToken=' + clientSideToken, {
+                    method: 'post',
+                    headers: {
+                        "Content-Type": "application/text",
+                        "Authorization": "Bearer " + clientSideToken
+                    },
+                    cache: 'default'
+                })
+                .then(response => response.json())
+                .then(data => {
+                        if (data.error == "consent_required") {
+                            setIsConsentButtonVisible(true);
+                            setIsLoginVisible(false);
+                            setLoading(false);
+                        }
+                        if (data) {
+                            return;
+                        } else {
+                            return;
```

Comment:
Why we are using if else condition, not meaning of this code.

Reply:
Yes , that If part can be avoided. Updated it